pkg/collector/corechecks/system/wincrashdetect/wincrashdetect.go

```diff
@@ -0,0 +1,200 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+//go:build windows
+
+package wincrashdetect
+
+import (
+	"fmt"
+
+	sysconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/system/wincrashdetect/probe"
+	dd_config "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/metrics/event"
+	process_net "github.com/DataDog/datadog-agent/pkg/process/net"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/retry"
+	"golang.org/x/sys/windows/registry"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const (
+	crashDetectCheckName = "wincrashdetect"
+	maxStartupWarnings   = 20
+)
+
+// Config is the configuration options for this check
+// it is exported so that the yaml parser can read it.
+type WinCrashConfig struct {
+	Enabled bool `yaml:"enabled"` // placeholder for config
+}
+
+type WinCrashDetect struct {
+	core.CheckBase
+	instance         *WinCrashConfig
+	hasRunOnce       bool
+	startupWarnCount int
+}
+
+func init() {
+	core.RegisterCheck(crashDetectCheckName, crashDetectFactory)
+}
+
+func crashDetectFactory() check.Check {
+	return &WinCrashDetect{
+		CheckBase:  core.NewCheckBase(crashDetectCheckName),
+		instance:   &WinCrashConfig{},
+		hasRunOnce: false,
+	}
+}
+
+// Parse parses the check configuration
+func (c *WinCrashConfig) Parse(data []byte) error {
+	// default values
+	c.Enabled = true
+
+	return yaml.Unmarshal(data, c)
+}
+
+func (wcd *WinCrashDetect) Configure(integrationConfigDigest uint64, config, initConfig integration.Data, source string) error {
+	err := wcd.CommonConfigure(integrationConfigDigest, initConfig, config, source)
+	if err != nil {
+		return err
+	}
+
+	return wcd.instance.Parse(config)
+}
+
+func (wcd *WinCrashDetect) handleStartupError(err error) error {
+	if retry.IsErrWillRetry(err) {
+		wcd.startupWarnCount++
+		// this is an expected error, the check run will occur before the system probe
+		// comes up.  However, it should resolve.  If the number of these exceeds a
+		// given threshold, report as an error anyway
+		if wcd.startupWarnCount > maxStartupWarnings {
+			return err
+		}
+		return nil
+	}
+	return err
+}
+func (wcd *WinCrashDetect) Run() error {
+
+	// only do this once; there's no point in doing it after the first one
+	if wcd.hasRunOnce {
+		log.Infof("check already run")
+		return nil
+	}
```

Comment:
I wonder if there is a more elegant mechanism to disable it in the scheduler

Reply:
there is not.